textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/RegexPredicate.java

```diff
@@ -1,88 +1,152 @@
-/**
- * 
- */
 package edu.uci.ics.textdb.dataflow.common;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.queryparser.classic.MultiFieldQueryParser;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.Query;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.FieldType;
 import edu.uci.ics.textdb.api.common.IField;
 import edu.uci.ics.textdb.api.common.IPredicate;
 import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.dataflow.ISourceOperator;
+import edu.uci.ics.textdb.api.storage.IDataStore;
+import edu.uci.ics.textdb.common.constants.DataConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
 import edu.uci.ics.textdb.common.field.Span;
-import edu.uci.ics.textdb.common.field.StringField;
-import edu.uci.ics.textdb.common.field.TextField;
+import edu.uci.ics.textdb.dataflow.regexmatch.RegexToGramQueryTranslator;
+import edu.uci.ics.textdb.dataflow.source.IndexBasedSourceOperator;
+import edu.uci.ics.textdb.storage.DataReaderPredicate;
 
 /**
+ * This class is responsible for calculating regex match. <br>
+ * It uses RegexToGramQueryTranslator to translate a regex to a query in lucene. <br>

```

Comment:
I agree with the reasoning. The regex matching logic should be in RegexMatcher, and the RegexPredicate should only contain 1: regex string, 2: list of attributes.

But now, since the predicate contains everything (attributes, analyzer, datastore), RegexPredicate essentially becomes RegexMatcher, and RegexMatcher does pretty much nothing except for calling getNextTuple(). 

So if we change it to that way, in regexMatcher we have to write
```
analyzer = regexPredicate.analyzer;
dataStore = regexPredicate.dataStore;
...
```
which seems to me is also bad.